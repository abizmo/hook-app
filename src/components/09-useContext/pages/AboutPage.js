import React, { useContext } from 'react'
import UserContext from '../context/UserContext'

const AboutPage = () => {
  const { user, setUser } = useContext(UserContext)

  const handleLogout = () => {
    setUser({})
  }

  return (
    <div>
      <h1>About Us</h1>
      <hr />

      <pre>
        {JSON.stringify(user, null, 3)}
      </pre>

      <button
        className="btn btn-danger"
        disabled={!user.id}
        onClick={handleLogout}
      >Logout</button>
    </div>
  )
}

export default AboutPage
