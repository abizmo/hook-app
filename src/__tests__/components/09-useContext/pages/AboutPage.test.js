import { mount } from "enzyme"

import UserContext from "../../../../components/09-useContext/context/UserContext"
import AboutPage from "../../../../components/09-useContext/pages/AboutPage"

describe('AboutPage test', () => {
  const user = {
    id: 123,
    name: 'abizmo',
    email: 'abizmo@abizmo.dev',
  }
  const setUser = jest.fn()
  const wrapper = mount(
    <UserContext.Provider value={{user, setUser}}>
      <AboutPage />
    </UserContext.Provider>
  )

  test('should render and match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('button').prop('disabled')).toBeFalsy()
  })

  test('should call setUser', () => {
    wrapper.find('button').prop('onClick')()

    expect(setUser).toHaveBeenCalledWith({})
  })
})
