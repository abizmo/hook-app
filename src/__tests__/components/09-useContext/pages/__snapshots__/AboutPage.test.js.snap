// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage test should render and match snapshot 1`] = `
<div>
  <h1>
    About Us
  </h1>
  <hr />
  <pre>
    {
   "id": 123,
   "name": "abizmo",
   "email": "abizmo@abizmo.dev"
}
  </pre>
  <button
    className="btn btn-danger"
    disabled={false}
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
